src/sentry/deletions/defaults/group.py

```diff
@@ -135,6 +152,12 @@ def delete_dangling_attachments_and_user_reports(self, events: Sequence[Event])
             event_id__in=event_ids, project_id__in=self.project_ids
         ).delete()
 
+    def delete_events_from_snuba(self) -> None:
```

Comment:
This is the logic from here:
<img width="1005" alt="image" src="https://github.com/user-attachments/assets/dee54807-604e-4d81-8c18-a3d702c2c18a">